zulip/integrations/clickup/zulip_clickup.py

```diff
@@ -0,0 +1,372 @@
+def redirect_to_clickup_auth(zulip_integration_url: str, client_id: str) -> None:
+    print(
+        """
+    STEP 2
+    ----
+    ClickUp authorization page will open in your browser.
+    Please authorize your workspace(s).
+
+    Click 'Connect Workspace' on the page to proceed...
+    """
+    )
+    parsed_url = urlparse(zulip_integration_url)
+    base_url: str = f"{parsed_url.scheme}://{parsed_url.netloc}"
+    url: str = f"https://app.clickup.com/api?client_id={client_id}&redirect_uri={base_url}"
+    time.sleep(1)
+    webbrowser.open(url)
+
+
+@clear_terminal_and_sleep(2)
+def query_for_authorization_code(zulip_integration_url: str) -> str:
+    print(
+        """
+    STEP 3
+    ----
+    After you've authorized your workspace,
+    you should be redirected to your home URL.
+    Please copy your home URL and paste it below.
+    It should contain a code, and look similar to this:
+
+    e.g. https://YourZulipDomain.com/?code=332KKA3321NNAK3MADS
+    """
+    )
+    input_url: str = input("YOUR HOME URL: ")
+    auth_code: str = process_url(input_url=input_url, base_url=zulip_integration_url)
+    return auth_code
+
+
+@clear_terminal_and_sleep(1)
+def query_for_notification_events() -> List[str]:
+    print(
+        """
+    STEP 4
+    ----
+    Please select which ClickUp event notification(s) you'd
+    like to receive in your Zulip app.
+    EVENT CODES:
+          1 = task
+          2 = list
+          3 = folder
+          4 = space
+          5 = goals
+
+    Here's an example input if you intend to only receive notifications
+    related to task, list and folder: 1,2,3
+    """
+    )
+    querying_user_input: bool = True
+    selected_events: List[str] = []
+
+    while querying_user_input:
+        input_codes: str = input("EVENT CODE(s): ")
+        user_input: List[str] = re.split(",", input_codes)
+
+        input_is_valid: bool = len(user_input) > 0
+        exhausted_options: List[str] = []
+
+        for event_code in user_input:
+            if event_code in EVENT_CHOICES and event_code not in exhausted_options:
+                selected_events += EVENT_CHOICES[event_code]
+                exhausted_options.append(event_code)
+            else:
+                input_is_valid = False
+
+        if not input_is_valid:
+            print("Please enter a valid set of options and only select each option once")
+
+        querying_user_input = not input_is_valid
+
+    return selected_events
+
+
+def delete_old_webhooks(zulip_integration_url: str, api_handler: ClickUpAPIHandler) -> None:
+    """
+    Checks for existing webhooks with the same endpoint and delete them if found.
+    """
+    data: Dict[str, Any] = api_handler.get_webhooks()
+    for webhook in data["webhooks"]:
+        print("webhook: ", webhook)
+        zulip_url_domain = urlparse(zulip_integration_url).netloc
+        registered_webhook_domain = urlparse(webhook["endpoint"]).netloc
+
+        if zulip_url_domain in registered_webhook_domain:
```

Comment:
This is done just to be conservative because we're registering and deleting webhook through ClickUp API calls. The user might have registered a webhook URL with additional query params such as stream and topic, and the user would have no way of deleting that one if they want to reconfigure their integration.